sycl/doc/extensions/proposed/sycl_ext_oneapi_usm_malloc_properties.asciidoc

```diff
@@ -0,0 +1,1615 @@
+|====
+
+
+The `sycl::ext::oneapi::experimental::usm_kind` property is a compile-time constant property with a single non-type parameter. This parameter is a value belonging to the enumeration `sycl::usm::alloc`.
+The `sycl::ext::oneapi::experimental::usm_kind` property is supported by `annotated_ptr` and the USM memory allocation functions defined in this extension.
+
+=== Deallocation
+
+This extension introduces two new functions called `sycl::ext::oneapi::experimental::free` that take an `annotated_ptr` as argument. These functions deallocate the memory pointed to by the raw pointer belonging to the `annotated_ptr`. The new deallocation functions are listed in Table <<table.usm.malloc.free>>. These functions belong to the namespace `sycl::ext::oneapi::experimental`.
+To avoid memory leaks, USM memory allocated using the USM memory allocation functions with `properties` support defined in this extension, must be deallocated using `sycl::ext::oneapi::experimental::free` or `sycl::free`.
+
+The following example shows how USM memory allocated using one of the functions defined in this extension can be deallocated, to avoid memory leaks.
+
+==== Example
+
+[source,c++]
+----
+using namespace sycl::ext::oneapi::experimental;
+
+// APtr and BPtr are of type
+// annotated_ptr<int*, decltype(properties{usm_kind<sycl::usm::alloc::device>})>
+auto APtr = annotated_malloc_device<int>(N, q);
+auto BPtr = annotated_malloc_device<int>(N, q);
+
+// Deallocate the memory pointed to by the raw pointer of APtr
+free(APtr, q);
+
+// Deallocate the memory pointed to by the raw pointer of BPtr, using sycl::free 
+free(BPtr.get(), q);
+----
+
+[[table.usm.malloc.free]]
+.New USM memory deallocation functions introduced by this extension
+[options="header"]
+|====
+|Function|Description
+a|
+[source,c++]
+----
+template<typename T, typename propList>
+void free(annotated_ptr<T, propList> &ptr,
+          const context& syclContext)
+----
+| Frees an allocation. The memory pointed to by the raw pointer belonging to `ptr` must have been allocated using one of the USM memory allocation functions with `properties` support defined in this extension, or one of the
+SYCL USM allocation routines. `syclContext` must be the same `context` that was used to allocate the memory. The memory is freed without waiting for `commands` operating on it to be completed. If `commands` that use this memory
+are in-progress or are enqueued the behavior is undefined. 
+a|
+[source,c++]
+----
+template<typename T, typename propList>
+void free(annotated_ptr<T, propList> &ptr,
+          const queue& syclQueue)
+----
+| Alternate form where `syclQueue` provides the `context`. 
+|====
+
+== Revision History
+
+[cols="5,15,15,70"]
+[grid="rows"]
+[options="header"]
+|========================================
+|Rev|Date|Author|Changes
+|1|2023-01-25|Jessica Davies|*Initial public working draft*
+|========================================
```

Comment:
We typically do not have a Revision History section because it usually does not get updated, and no on can agree on which changes merit an update to the history.  It's also confusing because each extension has an API revision identified by its feature-test macro.

Reply:
Removed.